assets/js/base/hooks/checkout/use-checkout-address.js

```diff
@@ -46,55 +43,54 @@ export const useCheckoutAddress = () => {
 			( ! customerId || isSameAddress( shippingAddress, billingData ) )
 	);
 
-	// Pushes to global state when changes are made locally.
-	useEffect( () => {
-		// Uses shipping address or billing fields depending on shippingAsBilling checkbox, but ensures
-		// billing only fields are also included.
-		const newBillingData = {
-			...( shippingAsBilling ? shippingAddress : billingFields ),
-			email: billingFields.email || billingData.email,
-			phone: billingFields.phone || billingData.phone,
-		};
-
-		if ( ! isEqual( newBillingData, billingData ) ) {
-			setBillingData( newBillingData );
-		}
-	}, [
-		billingFields,
-		shippingAsBilling,
-		billingData,
-		shippingAddress,
-		setBillingData,
-		setShippingAddress,
-	] );
+	const currentShippingAsBilling = useRef( shippingAsBilling );
```

Comment:
I'm not sure about this variable name. Is it really current as of the execution time? We typically use the `previous` prefix for `useRef` references (which is how it appears to be used in the logic here).

Reply:
This is current as in the context of `useCheckoutAddress` — if you see the return value from this hook it uses the `currentShippingAsBilling ` ref rather than `shippingAsBilling `. The reason I had to do it this way was because our billing address fields contain debounce code on the values, so if you return `shippingAsBilling `, those fields are rendered with the old values before this hook has had a chance to sync billing data.

I will add a comment to that effect if it helps.